lib/services/auth-service.js

```diff
@@ -0,0 +1,149 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+var jwt = require('jsonwebtoken');
+var crypto = require('crypto');
+
+module.exports = authServiceFactory;
+di.annotate(authServiceFactory, new di.Provide('Auth.Services'));
+di.annotate(authServiceFactory,
+    new di.Inject(
+        'Assert',
+        'Services.Configuration'
+    )
+);
+
+function authServiceFactory(
```

Comment:
@cgx027 Could you elaborate on why you chose crypto for this?  As far as I can tell, its only advantage is that it's faster which shouldn't matter in this application since the password hashing is only used during login.  The justification mentioned in google groups seems to be just to avoid bringing in a new module.